app/src/main/java/com/android/streetworkapp/model/park/ParkRepositoryFirestore.kt

```diff
@@ -0,0 +1,303 @@
+    require(pid.isNotEmpty()) { PID_EMPTY }
+    require(capacity > 0) { "Capacity must be greater than 0." }
+    try {
+      db.collection(COLLECTION_PATH).document(pid).update("capacity", capacity).await()
+    } catch (e: Exception) {
+      Log.e("FirestoreError", "Error updating capacity from park: ${e.message}")
+    }
+  }
+
+  /**
+   * Increment the occupancy of a park by one person.
+   *
+   * @param pid The park ID.
+   */
+  override suspend fun incrementOccupancy(pid: String) {
+    require(pid.isNotEmpty()) { PID_EMPTY }
+    try {
+      val document = db.collection(COLLECTION_PATH).document(pid).get().await()
+      val currentOccupancy = document.getLong("occupancy")?.toInt() ?: 0
+
+      db.collection(COLLECTION_PATH).document(pid).update("occupancy", currentOccupancy + 1).await()
+    } catch (e: Exception) {
+      Log.e("FirestoreError", "Error incrementing occupancy for park: ${e.message}")
+    }
+  }
+
+  /**
+   * Decrement the occupancy of a park by one person.
+   *
+   * @param pid The park ID.
+   */
+  override suspend fun decrementOccupancy(pid: String) {
+    require(pid.isNotEmpty()) { PID_EMPTY }
+    try {
+      val document = db.collection(COLLECTION_PATH).document(pid).get().await()
+      val currentOccupancy = document.getLong("occupancy")?.toInt() ?: 0
+
+      val newOccupancy = if (currentOccupancy > 0) currentOccupancy - 1 else 0
+
+      db.collection(COLLECTION_PATH).document(pid).update("occupancy", newOccupancy).await()
+    } catch (e: Exception) {
+      Log.e("FirestoreError", "Error decrementing occupancy for park: ${e.message}")
+    }
+  }
+
+  /**
+   * Add an event to a park.
+   *
+   * @param pid The park ID.
+   * @param eid The event ID.
+   */
+  override suspend fun addEventToPark(pid: String, eid: String) {
+    require(pid.isNotEmpty()) { PID_EMPTY }
+    require(eid.isNotEmpty()) { "Event ID cannot be empty." }
+    try {
+      db.collection(COLLECTION_PATH)
+          .document(pid)
+          .update("events", FieldValue.arrayUnion(eid))
+          .await()
+    } catch (e: Exception) {
+      Log.e("FirestoreError", "Error adding event to park: ${e.message}")
+    }
+  }
+
+  /**
+   * Delete an event from a park.
+   *
+   * @param pid The park ID.
+   * @param eid The event ID.
+   */
+  override suspend fun deleteEventFromPark(pid: String, eid: String) {
+    require(pid.isNotEmpty()) { PID_EMPTY }
+    require(eid.isNotEmpty()) { "Event ID cannot be empty." }
+    try {
+      db.collection(COLLECTION_PATH)
+          .document(pid)
+          .update("events", FieldValue.arrayRemove(eid))
+          .await()
+    } catch (e: Exception) {
+      Log.e("FirestoreError", "Error deleting event from park: ${e.message}")
+    }
+  }
+
+  /**
+   * Delete a park by its ID.
+   *
+   * @param pid The park ID.
+   */
+  override suspend fun deleteParkByPid(pid: String) {
+    require(pid.isNotEmpty()) { PID_EMPTY }
```

Comment:
pid.isNotEmpty() is used a lot in the code.
Maybe making it into a function can better the readability and consistency

Reply:
I can see your idea, but `require` only takes one line of code, so it would be pointless to declare a function and call it each time since it would also take only one line, what do you think?

Reply:
Maybe the function would also be practical in the future if you need to add more condition like these.
But yes, right now it's not inconvenient and doesnt change much